apps/web/src/pages/ApplicantDashboardPage/components/DashboardHeading.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from "react";
+import { useIntl } from "react-intl";
+import {
+  BriefcaseIcon,
+  BookOpenIcon,
+  UsersIcon,
+  LightBulbIcon,
+  StarIcon,
+} from "@heroicons/react/24/solid";
+
+import { User } from "@gc-digital-talent/graphql";
+
+import Hero from "~/components/Hero/Hero";
+import useRoutes from "~/hooks/useRoutes";
+import { HeroCard } from "./HeroCard";
+import {
+  HeroCardExperienceItem,
+  HeroCardProfileItem,
+  ProfileItemStatus,
+} from "./HeroCardItem";
+import {
+  hasEmptyRequiredFields as aboutSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as aboutSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/AboutSection";
+import {
+  hasEmptyRequiredFields as languageInformationSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as languageInformationSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/LanguageInformationSection";
+import {
+  hasEmptyRequiredFields as governmentInformationSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as governmentInformationSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/GovernmentInformationSection";
+import {
+  hasEmptyRequiredFields as workLocationSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as workLocationSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/WorkLocationSection";
+import {
+  hasEmptyRequiredFields as workPreferencesSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as workPreferencesSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/WorkPreferencesSection";
+import {
+  hasEmptyRequiredFields as diversityEquityInclusionSectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as diversityEquityInclusionSectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/DiversityEquityInclusionSection";
+import {
+  hasEmptyRequiredFields as roleSalarySectionHasEmptyRequiredFields,
+  hasEmptyOptionalFields as roleSalarySectionHasEmptyOptionalFields,
+} from "../../../components/UserProfile/ProfileSections/RoleSalarySection";
+
+export interface DashboardHeadingProps {
+  user: User;
+}
+
+export const DashboardHeading = ({ user }: DashboardHeadingProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+
+  function deriveSectionStatus(
+    isMissingRequiredFields: boolean,
+    isMissingOptionalFields: boolean,
+  ): ProfileItemStatus {
+    if (isMissingRequiredFields) return "has-empty-required-fields";
+    if (isMissingOptionalFields) return "has-empty-optional-fields";
+    return "all-sections-complete";
+  }
```

Comment:
If it is possible, can we move this outside of the component function so it doesn't get defined on every render?

Reply:
good idea - [move function out of component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5857/commits/57847b642b8e264c063f73ca956673fe3da24f6d)